tracer/test/Datadog.Trace.ClrProfiler.Managed.Tests/SourceLinkUriParserTests.cs

```diff
@@ -0,0 +1,72 @@
+                yield return new object[] { "https://api.bitbucket.org/2.0/repositories/test-org/test-repo/src/dd35903c688a74b62d1c6a9e4f41371c65704db8/*", "dd35903c688a74b62d1c6a9e4f41371c65704db8", "https://bitbucket.org/test-org/test-repo", typeof(BitBucketSourceLinkUrlParser) };
+                yield return new object[] { "https://test.visualstudio.com/test-org/_apis/git/repositories/my-repo/items?api-version=1.0&versionType=commit&version=dd35903c688a74b62d1c6a9e4f41371c65704db8&path=/*", "dd35903c688a74b62d1c6a9e4f41371c65704db8", "https://test.visualstudio.com/test-org/_git/my-repo", typeof(AzureDevOpsSourceLinkUrlParser) };
+                yield return new object[] { "https://test-gitlab-domain.com/test-org/test-repo/raw/dd35903c688a74b62d1c6a9e4f41371c65704db8/*", "dd35903c688a74b62d1c6a9e4f41371c65704db8", "https://test-gitlab-domain.com/test-org/test-repo", typeof(GitLabSourceLinkUrlParser) };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidTestCases))]
+        public void ProviderSpecificParsers_ParsesSuccessfully(string url, string expectedSha, string expectedRepositoryUrl, Type parserType)
+        {
+            var parser = (SourceLinkUrlParser)Activator.CreateInstance(parserType)!;
+
+            var result = parser.ParseSourceLinkUrl(new Uri(url), out var sha, out var repositoryUrl);
+
+            result.Should().BeTrue();
+            sha.Should().Be(expectedSha);
+            repositoryUrl.Should().Be(expectedRepositoryUrl);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidTestCases))]
+        [SuppressMessage("Usage", "xUnit1026:Theory methods should use all of their parameters", Justification = "Testing the general implementation")]
+        public void CompositeParser_ParsesSuccessfully(string url, string expectedSha, string expectedRepositoryUrl, Type parserType)
+        {
+            var result = CompositeSourceLinkUrlParser.Instance.ParseSourceLinkUrl(new Uri(url), out var sha, out var repositoryUrl);
+
+            result.Should().BeTrue();
+            sha.Should().Be(expectedSha);
+            repositoryUrl.Should().Be(expectedRepositoryUrl);
+        }
+
+        [Theory]
+        [InlineData("https://raw.githubusercontent.com/DataDog/dd-trace-dotnet/invalid-sha/*")]
+        [InlineData("https://raw.githubusercontent.com/DataDog/dd-trace-dotnet/dd35903c688a74b62d1c6a9e4f41371c65704db8/extraneous/stuff/that/shouldnt/be/here/*")]
+        [InlineData("https://raw.githubusercontent.com/DataDog/dd-trace-dotnet/*")]
+        [InlineData("https://api.bitbucket.org/2.0/repositories/test-org/test-repo/src/*")] // missing sha
+        [InlineData("https://test.visualstudio.com/test-org/_apis/git/repositories/my-repo/items?api-version=1.0&versionType=commit&path=/*")] // missing sha
+        [InlineData("https://test-gitlab-domain.com/test-org/test-repo/raw/*")] // missing sha
+        [InlineData("https://test.com/test-org/test-repo/raw/invalid-sha/*")] // missing sha
```

Comment:
Would be nice to see more malformed urls IMO to get extra coverage of the parsers. This is one of the cases where the parsers should be 100% branch covered IMO :)

Reply:
Added a few more test cases in 233cf69448d27b82b4c1721ca90bb74fab58032f. Let me know if you have any more ideas on what to test.